src/cascadia/UnitTests_Control/ControlInteractivityTests.cpp

```diff
@@ -0,0 +1,341 @@
+
+        Log::Comment(L"Scroll in the positive direction, increasing opacity");
+        // Scroll more than enough times to get to 1.0 from .5.
+        for (int i = 0; i < 55; i++)
+        {
+            // each mouse wheel only adjusts opacity by .01
+            expectedOpacity += 0.01;
+            if (expectedOpacity >= 1.0)
+            {
+                expectedOpacity = 1.0;
+            }
+
+            // The mouse location and buttons don't matter here.
+            interactivity->MouseWheel(modifiers,
+                                      30,
+                                      til::point{ 0, 0 },
+                                      { false, false, false });
+        }
+
+        Log::Comment(L"Scroll in the negative direction, decreasing opacity");
+        // Scroll more than enough times to get to 0.0 from 1.0
+        for (int i = 0; i < 105; i++)
+        {
+            // each mouse wheel only adjusts opacity by .01
+            expectedOpacity -= 0.01;
+            if (expectedOpacity <= 0.0)
+            {
+                expectedOpacity = 0.0;
+            }
+
+            // The mouse location and buttons don't matter here.
+            interactivity->MouseWheel(modifiers,
+                                      30,
+                                      til::point{ 0, 0 },
+                                      { false, false, false });
+        }
+    }
+
+    void ControlInteractivityTests::TestPanWithTouch()
+    {
+        VERIFY_IS_TRUE(false);
+    }
+    void ControlInteractivityTests::TestScrollWithMouse()
+    {
+        WEX::TestExecution::DisableVerifyExceptions disableVerifyExceptions{};
+
+        auto [settings, conn] = _createSettingsAndConnection();
+        auto [core, interactivity] = _createCoreAndInteractivity(*settings, *conn);
+        _standardInit(core, interactivity);
+        // For the sake of this test, scroll one line at a time
+        interactivity->_rowsToScroll = 1;
+
+        int expectedTop = 0;
+        int expectedViewHeight = 20;
+        int expectedBufferHeight = 20;
+
+        auto scrollChangedHandler = [&](auto&&, const Control::ScrollPositionChangedArgs& args) mutable {
+            VERIFY_ARE_EQUAL(expectedTop, args.ViewTop());
+            VERIFY_ARE_EQUAL(expectedViewHeight, args.ViewHeight());
+            VERIFY_ARE_EQUAL(expectedBufferHeight, args.BufferSize());
+        };
+        core->ScrollPositionChanged(scrollChangedHandler);
+        interactivity->ScrollPositionChanged(scrollChangedHandler);
+
+        for (int i = 0; i < 40; ++i)
+        {
+            Log::Comment(NoThrowString().Format(L"Writing line #%d", i));
+            // The \r\n in the 19th loop will cause the view to start moving
+            if (i >= 19)
+            {
+                expectedTop++;
+                expectedBufferHeight++;
+            }
+
+            conn->WriteInput(L"Foo\r\n");

```

Comment:
we're using the mock connection's input method to get lines back out through the terminal. this is _definitely_ more of a functional test than a unit test

Reply:
shhhhhh....

we're already using "unit test" to mean "thing that can run easily in CI without a HWND". The screen buffer tests, the _roundtrip tests_, those are all definitely pushing the boundary of "unit test"